src/genai/services/request_handler.py

```diff
@@ -295,16 +296,17 @@ def patch(endpoint: str, key: str, json_data: dict) -> Response:
 
     @staticmethod
     def post_stream(endpoint, headers, json_data, files):
-        with HttpProvider.get_client(timeout=ConnectionManager.TIMEOUT) as s:
-            with s.stream(
+        with HttpProvider.get_client(timeout=ConnectionManager.TIMEOUT) as client:
+            with connect_sse(
+                client,
                 method="POST",
                 url=endpoint,
                 headers=headers,
                 json=json_data,
                 files=files,
-            ) as r:
-                for chunk in r.iter_text():
-                    yield chunk
+            ) as event_source:
+                for sse in event_source.iter_sse():
+                    yield sse.data
```

Comment:
The old version has contained a very naive approach to parsing SSE as a plain string.
`httpx_sse` handles out-of-the-box chunk parsing and mainly network errors. With the old approach, some messages got lost because of invalid parsing.